### PR TITLE
8290211: jdk/internal/vm/Continuation/Fuzz.java failed with "AssertionError: Failed to compile int Fuzz.com_int(int,int) in 5000ms"

### PR DESCRIPTION
**Pull request by dcubed-ojdk**
A trivial fix so that Continuation/Fuzz.java honors the timeoutFactor JTREG setting
when waiting for a compilation to finish.

This fix is being tested in my jdk-20+10 stress testing run.

The usual Mach5 timeoutFactor is 4.0 with slower configurations using a timeoutFactor
of 10.0. In my stress testing, I use release-bits: 4.0, fastdebug-bits: 6.0 and slowdebug-bits: 12.0.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8290211](https://bugs.openjdk.org/browse/JDK-8290211): jdk/internal/vm/Continuation/Fuzz.java failed with "AssertionError: Failed to compile int Fuzz.com_int(int,int) in 5000ms"

### Reviewers
 * [Leonid Mesnik](https://openjdk.org/census#lmesnik) (@lmesnik - **Reviewer**) ⚠️ Review applies to [bd9be934](https://git.openjdk.org/jdk/pull/9844/files/bd9be9348e98d1f9052024f9075d7b32c6c85c20)
 * [Alan Bateman](https://openjdk.org/census#alanb) (@AlanBateman - **Reviewer**) ⚠️ Review applies to [bd9be934](https://git.openjdk.org/jdk/pull/9844/files/bd9be9348e98d1f9052024f9075d7b32c6c85c20)
 * [Jie Fu](https://openjdk.org/census#jiefu) (@DamonFool - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/9844/head:pull/9844` \
`$ git checkout pull/9844`

Update a local copy of the PR: \
`$ git checkout pull/9844` \
`$ git pull https://git.openjdk.org/jdk pull/9844/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 9844`

View PR using the GUI difftool: \
`$ git pr show -t 9844`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/9844.diff">https://git.openjdk.org/jdk/pull/9844.diff</a>

</details>

### CODE REVIEW

**Review by lmesnik — APPROVED** [review]

File: test/jdk/jdk/internal/vm/Continuation/Fuzz.java [hunk 1]
```diff
@@ -78,7 +78,8 @@ public class Fuzz implements Runnable {
     static final boolean RANDOM  = true;
     static final boolean VERBOSE = false;
 
-    static final int COMPILATION_TIMEOUT = 5_000; // ms
+    static float timeoutFactor = Float.parseFloat(System.getProperty("test.timeout.factor", "1.0"));
```
Comment (DamonFool) [inline test/jdk/jdk/internal/vm/Continuation/Fuzz.java:81]: How about something like this?
```
System.getProperty("test.timeout.factor", "5.0")
```

Comment (DamonFool) [reply test/jdk/jdk/internal/vm/Continuation/Fuzz.java:81]: Or

```
static final int COMPILATION_TIMEOUT = max((int)(1_000 * timeoutFactor), 5_000); // ms
```

**Review by AlanBateman — APPROVED** [review]

File: test/jdk/jdk/internal/vm/Continuation/Fuzz.java [hunk 2]
```diff
@@ -78,7 +78,8 @@ public class Fuzz implements Runnable {
...
+    static final int COMPILATION_TIMEOUT = (int)(1_000 * timeoutFactor); // ms
```
Comment (dcubed-ojdk) [inline test/jdk/jdk/internal/vm/Continuation/Fuzz.java:82]: I was really hoping that `1_000 * timeoutFactor` would do the trick,
but stress run-#1 on my macosx-aarch64 machine had a couple of
sub-test timeouts at 12 seconds (1_000 * 12.0). My timeoutFactor
for slowdebug is 12.0. My linux-x64 machine passed stress run-#1
without any failures.

I've switched the value from 1_000 back to 5_000 for the next two
stress runs on both machines.

File: test/jdk/jdk/internal/vm/Continuation/Fuzz.java (hunk 1 above)
Comment (dcubed-ojdk) [reply test/jdk/jdk/internal/vm/Continuation/Fuzz.java:81]: > System.getProperty("test.timeout.factor", "5.0")
Definitely not. Traditional default value for no specified timeoutFactor value is 1.0.

> static final int COMPILATION_TIMEOUT = max((int)(1_000 * timeoutFactor), 5_000); // ms
So you're trying to make sure we have a minimum COMPILATION_TIMEOUT value of 5 seconds,
but I'm not sure why you want that.

```
static final int COMPILATION_TIMEOUT = (int)(5_000 * timeoutFactor); // ms
```

i.e., a `default value * timeoutFactor` is the usual way to do this. I've seen some
folks use a timeoutFactor of 0.5 on really fast machines, but that wouldn't work
with your minimum COMPILATION_TIMEOUT value of 5 seconds above.

**Review by dcubed-ojdk — COMMENTED** [review]
@lmesnik, @DamonFool and @AlanBateman - Thanks for the reviews.

**Comment by dcubed-ojdk** [comment]
@lmesnik, @DamonFool and @AlanBateman - Thanks for the reviews.

**Comment by DamonFool** [comment]
> So you're trying to make sure we have a minimum COMPILATION_TIMEOUT value of 5 seconds,
but I'm not sure why you want that.

With the current patch, `COMPILATION_TIMEOUT` may become less than 5_000, which may lead to more failures on various platforms.
That's why I suggest let `COMPILATION_TIMEOUT >= 5_000`, which won't make things worse.

Thanks.

**Comment by DamonFool** [comment]
> Definitely not. Traditional default value for no specified timeoutFactor value is 1.0.

Okay, you're right.
This is because the timeoutFactor may be used by other tests too.

**Comment by dcubed-ojdk** [comment]
Sorry for the delay in getting back to this PR. I've been focused on GateKeeping issues instead.

This latest update:

- Scale the original COMPILATION_TIMEOUT value of 5 seconds by timeoutFactor.

makes the test happy on my linux-x64 stress runs. My macosx-aarch64 stress runs
are still not happy, but I'm still gathering data on those runs. I haven't found a
COMPILATION_TIMEOUT initial value that works every time yet. The values I've
tried so far:
- 1_000 - the original value in the PR
- 5_000 - the original value in the test before I changed it
- 10_000 - simple doubling
- 20_000 - simple doubling again, testing this value now and thru the weekend

I'm inclined to move ahead with the 5_000 value scaled by timeoutFactor. I think
I need to spend some time to determine why macosx-aarch64 is so much slower
than linux-x64 for this test.

**Review by DamonFool — APPROVED** [review]
LGTM

Thanks for the update.

**Comment by dcubed-ojdk** [comment]
@DamonFool - Thanks for the re-review!

**Comment by dcubed-ojdk** [comment]
jdk/internal/vm/Continuation/Fuzz.java* passed 12 times (2 sub-tests x 6 configs)
in two different Mach5 Tier1 job sets.

**Pull request closed without merging**